hpx/runtime/serialization/variant.hpp

```diff
@@ -0,0 +1,78 @@
+//  Copyright (c) 2015 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_VARIANT_HPP
+#define HPX_SERIALIZATION_VARIANT_HPP
+
+#include <hpx/runtime/serialization/serialize.hpp>
+
+#include <boost/variant.hpp>
+
+namespace hpx { namespace serialization
+{
+    namespace detail
+    {
+        struct variant_save_visitor: boost::static_visitor<>
+        {
+            variant_save_visitor(output_archive& ar)
+              : ar(ar)
+            {}
+
+            template <class T>
+            void operator()(const T& value)
+            {
+                ar << value;
+            }
+
+        private:
+            output_archive& ar;
+        };
+
+        template <class... Args>
+        struct load_helper_t{};
+
+        template <class Variant, class... Tail>
+        BOOST_FORCEINLINE void load_helper(input_archive& ar, Variant& var,
+                int which, load_helper_t<boost::detail::variant::void_, Tail...>)
```

Comment:
The macros are `BOOST_VARIANT_ENUM_PARAMS` and `BOOST_VARIANT_ENUM_SHIFTED_PARAMS`. It seems they are only partially documented [here](http://www.boost.org/doc/libs/1_58_0/doc/html/variant/tutorial.html#variant.tutorial.preprocessor), so try the examples and tests to see how they are used.